snap/hooks/install

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+
+set -ex
+
+# Copy Spack Config
+# - $SNAP_COMMON/config.yaml       <-- typically /var/snap/spack/common/config.yaml
+# env; cp $SNAP/config.yaml $SNAP_COMMON/config.yaml
+SPACK=/opt/$SNAP_NAME
+
+ if [ ! -d "$SPACK" ]; then
+    mkdir $SPACK
+    chmod 777 $SPACK
```

Comment:
Hmm... We definitely do not want to be giving everyone on the system rwx privileges to _/opt/spack_. This will create a recipe for disaster as anyone can go into this directory and decide "you know what, nobody gets Fortran now. Use a newer language boomer!" The privileges should be set to 755 so that only root can write to that location, but users outside root can still read the module files and binaries:

```suggestion
    chmod 755 $SPACK
```

Reply:
Ah yeah, for my hacking I gave it full permissions but shouldn't be in the final snap. Tested bringing it down to 755 makes more sense and still works fine.